packages/backend/src/database/migrations/20220215144117_add_dashboard_views_table.ts

```diff
@@ -0,0 +1,49 @@
+import { Knex } from 'knex';
+
+const DashboardViewsTableName = 'dashboard_views';
+const DashboardVersionsTableName = 'dashboard_versions';
+
+export async function up(knex: Knex): Promise<void> {
+    if (!(await knex.schema.hasTable(DashboardViewsTableName))) {
+        await knex.schema.createTable(DashboardViewsTableName, (table) => {
+            table.specificType(
+                'dashboard_view_id',
+                `integer PRIMARY KEY GENERATED ALWAYS AS IDENTITY`,
+            );
+            table
+                .uuid('dashboard_view_uuid')
+                .notNullable()
+                .defaultTo(knex.raw('uuid_generate_v4()'));
+            table
+                .timestamp('created_at', { useTz: false })
+                .notNullable()
+                .defaultTo(knex.fn.now());
+            table.text('name').notNullable();
+            table
+                .integer('dashboard_version_id')
+                .notNullable()
+                .references('dashboard_version_id')
+                .inTable(DashboardVersionsTableName)
+                .onDelete('CASCADE');
+            table.jsonb('filters').notNullable();
+        });
+    }
+    if (await knex.schema.hasTable(DashboardVersionsTableName)) {
```

Comment:
Is it really essential to have an entry for existing dashboards? I imagined dashboards having zero->many filter views?

Reply:
I thought making sure there is always a default view would make our lives easier and code cleaner. 🤔  Even if the default view had an empty **DashboardFilters** object.  

As a user, when I create a dashboard, I expect to have a default view. We could have some FE magic to make them believe they have a default view when they don't, but ... that brings extra logic to our code. 

Let me know your thoughts